web-api/src/models.rs

```diff
@@ -182,26 +182,31 @@ pub(crate) struct DocumentProperty(Value);
 
 impl DocumentProperty {
     pub(crate) fn try_from_value(
-        document_id: &DocumentId,
         property_id: &DocumentPropertyId,
         mut value: Value,
     ) -> Result<Self, InvalidDocumentProperty> {
+        let check_string_property_value = |s: &str| {
+            if s.len() <= 2_048 && GENERIC_STRING_SYNTAX.is_match(s) {
+                Ok(())
+            } else {
+                Err(())
+            }
+        };
+
         match &mut value {
             Value::Bool(_) | Value::Number(_) | Value::Null => {}
             Value::String(string) => {
-                if !is_valid_string_empty_ok(string, 2_048) {
+                let Ok(()) = check_string_property_value(string) else {
```

Comment:
:wrench: a `if is_err()` seems to transport the intention a bit better than `let Ok(())`, with the reused `validate_string()` we can write it as
```suggestion
                if validate_string(string).is_err() {
```